### PR TITLE
[UPDATE] Support to cache images

### PR DESCRIPTION
**Pull request by rogerioefonseca**
Add support to cache images, and load then from memory instead of disk.
Used zend-cache 2 factory libray backend

### CODE REVIEW

File: src/visualCaptcha/Captcha.php [hunk 1]
```diff
@@ -254,7 +276,7 @@ private function utilReadJSON( $filePath ) {
     }
 
     // Stream file from path
-    private function utilStreamFile( &$headers, $filePath ) {
+    private function utilStreamFile( $headers, $filePath ) {
```
Comment (BrunoBernardino) [inline src/visualCaptcha/Captcha.php]: By removing the pointer here we no longer change the headers, so we'd have to return them, or they'd have to be set somewhere else, right? Or am I missing something?

File: README.md [hunk 2]
```diff
@@ -37,6 +37,26 @@ Where:
 
 - `$namespace` is optional. It's a string and defaults to 'visualcaptcha'. You'll need to specifically set this if you're using more than one visualCaptcha instance in the same page, so the code can identify from which one is the validation coming from.
 
+### Using Cache
+
+You can use a backend Zend-Cache library options to store images on cache and avoid I/O.
+You'll have to pass options parameter on the cosntructor as document in https://docs.zendframework.com/zend-cache/storage/adapter/.
```
Comment (BrunoBernardino) [inline README.md]: Small typo: `cosntructor` vs `constructor`

File: README.md [hunk 3]
```diff
@@ -37,6 +37,26 @@ Where:
...
+By dafault it is disabled and you�ll have to pass true on the constructor on the 5th parameter.
```
Comment (BrunoBernardino) [inline README.md]: Another small typo: `dafault` vs `default`

File: composer.json [hunk 4]
```diff
@@ -17,7 +17,8 @@
         "issues": "https://github.com/emotionLoop/visualCaptcha-packagist/issues"
     },
     "require": {
-        "php": ">=5.3"
+        "php": ">=5.3",
+        "zendframework/zend-cache" : "<=2.0.3"
```
Comment (BrunoBernardino) [inline composer.json]: Is this really _below_ 2.0.3 or is it a typo?

File: tests/unit/CustomAssetsTest.php [hunk 5]
```diff
@@ -14,8 +15,33 @@ public function testCustomAssets() {
...
+                        ),
+                    );
+        
+        
+        $captchaWithCache = new \visualCaptcha\Captcha( $this->session,$assetsPath,null,null, true,$options);
+        $imageOptions = $captchaWithCache->getAllImageOptions();
+        $this->assertCount( 2, $imageOptions );
+        $this->assertEquals( 'Cat', $imageOptions[ 0 ][ 'name' ] );
+        
+        $captchaWithouCache = new \visualCaptcha\Captcha( $this->session,$assetsPath);
```
Comment (BrunoBernardino) [inline tests/unit/CustomAssetsTest.php]: Small typo:`$captchaWithouCache` vs `$captchaWithoutCache`.

Also, `$imageOptions` below isn't being fetched from this captcha to test, it's asserting the same thing above.

**Comment by rogerioefonseca** [comment]
Awsering you question.
Zend-Cache is about 1.1M.
There are others as http://www.stashphp.com/ but i really like to use Zend because of the support and maintenance of the community and company.

File: README.md (hunk 2 above)
Comment (rogerioefonseca) [inline README.md]: Hello it is 0.0.7 on your master so... 
Is it really to downgrade it? 
Or.. do i update it to 0.0.8?
Or.. do i keep ir on 0.0.7?

Comment (BrunoBernardino) [inline README.md]: Sorry, update it to `0.0.8`. I looked at your fork before, not master.

**Comment by BrunoBernardino** [comment]
Thank you so much for the changes @rogerioefonseca !!

**Pull request merged**